units/L_Ghast.cfg

```diff
@@ -2,41 +2,20 @@
 
 [unit_type]
     id=L Ghast
+    [base_unit]
+        id=Ghast
+    [/base_unit]
     name= _ "Cursed Ghast"
-    race=undead
-    {TRAIT_FEARLESS_MUSTHAVE}
-    image="units/ghast2/ghast2.png"
-    profile="portraits/undead/ghoul.png"
-    {MAGENTA_IS_THE_TEAM_COLOR}
-    hitpoints=65
-    movement_type=gruefoot
-    movement=5
-    experience=150
-    level=3
-    alignment=chaotic
-    advances_to=null
-    {AMLA_DEFAULT}
-    cost=41
-    usage=fighter
     description= _ "This ghast is put under a curse by Ancient Lich. As a result, its poison can work against even undead units."+{SPECIAL_NOTES}+{SPECIAL_NOTES_POISON}+{SPECIAL_NOTES_FEEDING}+{SPECIAL_NOTES_REGENERATES}+{SPECIAL_NOTES_ARCANE}
-    die_sound=ghoul-hit.wav
     [abilities]
         {ABILITY_FEEDING}
         {ABILITY_REGENERATES}
     [/abilities]
-    {DEFENSE_ANIM "units/ghast2/ghast2-defend-2.png" "units/ghast2/ghast2-defend-1.png" ghoul-hit.wav }
+
+    # because of [base_unit] we need this, otherwise we change the first attack
     [attack]
-        name=bite
-        description={STR_BITE}
-        icon=attacks/fangs.png
-        type=blade
-        range=melee
-        damage=10
-        number=3
-        [specials]
-            {WEAPON_SPECIAL_POISONG}
-        [/specials]
```

Comment:
The problem is, you can't remove things with [base_unit].
The weapon special macros are in the end all different tags. You could change things in the tag, e.g. the % in [chance_to_hit], but not the whole tag.

In your case

[attack]
[/attack]

[attack]
[/attack]

  [attack]
    name=staff
    description={STR_STAFF}
    icon=attacks/staff-ruby.png
    type=impact
    range=melee
    damage=5
    number=2
  [/attack]

would work, but have the attack at the end.

This isn't quite ideal, I would like to propose a change to [base_unit], but I don't really know how it should be.